src/test/datascience/kernel-launcher/kernelFinder.vscode.test.ts

```diff
@@ -37,16 +37,31 @@ suite('DataScience - Kernels Finder', () => {
         });
         assert.isUndefined(kernelSpec);
     });
-    test('Python kernel returned if no matching kernel found', async () => {
+    test('Python kernel should not be returned if no matching kernel found (for notebooks)', async () => {
+        const kernelSpec = await kernelFinder.findKernel(resourceToUse, {
+            kernelspec: { display_name: 'foobar', name: 'foobar' },
+            orig_nbformat: 4
+        });
+        assert.isUndefined(kernelSpec);
+    });
+    test('Python kernel should be returned if no matching kernel found (for interactive window)', async () => {
```

Comment:
For interactive window we always return a python kernel (even if we don't find an exact match)